### PR TITLE
[US1, US2]: add an icon to allow users to add more files into the application

### PR DESCRIPTION
**Pull request by liviaw**
This PR consists of the following functionalities:
* Loading bar to show that app is rendering preview files
* Add more files

changes in code:
* Loading folder for a loading bar
* AddMediaIcon folder shows an svg and have onclick handler to attach more files with checker

Screen Shot 2021-01-14 at 3 56 32 pm

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -5,6 +5,10 @@ import { ImportModal, Header } from "./Components/";
 const App: React.FC = () => {
   const [files, setFiles] = useState<File[]>([]);
   const [show, setShow] = useState<boolean>(false);
+  const [totalVideoDuration, setTotalVideoDuration] = useState<number>(0);
+  // mapping of file index to original duration of video/images
+  const [oriDur, setOriDur] = useState<{ [fileindex: number]: number }>({});
```
Comment (juliamitchelmore) [inline src/App.tsx]: This information should live alongside all the file data. Again, TBD in our follow-up conversation (mentioned in #6 )

File: src/Components/AddMediaIcon/AddMediaIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+type Props = {
+  createMediaElement: (files: File[]) => void;
+};
+
+export const AddMediaIcon: React.FC<Props> = ({ createMediaElement }) => {
+  const handleVideoUpload = async (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    if (event.target.files === null) return;
+    let files: FileList = event.target.files;
```
Comment (juliamitchelmore) [inline src/Components/AddMediaIcon/AddMediaIcon.tsx]: This is a `const` as 'files' doesn't get mutated

File: src/Components/AddMediaIcon/AddMediaIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    const addedFiles: File[] = [];
+    for (let i = 0; i < files.length; i++) {
+      if (isImage(files[i]) || isVideo(files[i])) {
+        addedFiles.push(files[i]);
+      } 
+    }
```
Comment (juliamitchelmore) [inline src/Components/AddMediaIcon/AddMediaIcon.tsx]: Can this be rewritten as a `map`?

File: src/Components/AddMediaIcon/photo.svg [hunk 4]
```diff
@@ -0,0 +1 @@
...4v-400a8 8 0 0 0 -8-8zm-80 480a72 72 0 1 1 72-72 72.083 72.083 0 0 1 -72 72zm72-122.54a87.974 87.974 0 0 0 -160 50.54c0 2.7.13 5.36.37 8h-304.37v-400h464z"/><path d="m464 40h-416a8 8 0 0 0 -8 8v256a8 8 0 0 0 8 8h416a8 8 0 0 0 8-8v-256a8 8 0 0 0 -8-8zm-376.29 256 64.62-90.19 71.53 90.19zm156.58 0-40.58-51.16 84.63-119.48 134.51 170.64zm211.71 0h-12.77l-148.95-188.95a7.99 7.99 0 0 0 -12.81.33l-88.12 124.4-35.08-44.22a8.084 8.084 0 0 0 -6.47-3.03 7.983 7.983 0 0 0 -6.3 3.34l-77.47 108.13h-12.03v-240h400z"/><path d="m160 160a40 40 0 1 0 -40-40 40.045 40.045 0 0 0 40 40zm0-64a24 24 0 1 1 -24 24 24.028 24.028 0 0 1 24-24z"/><path d="m440 392v-24a8 8 0 0 0 -8-8h-32a8 8 0 0 0 -8 8v24h-24a8 8 0 0 0 -8 8v32a8 8 0 0 0 8 8h24v24a8 8 0 0 0 8 8h32a8 8 0 0 0 8-8v-24h24a8 8 0 0 0 8-8v-32a8 8 0 0 0 -8-8zm16 32h-24a8 8 0 0 0 -8 8v24h-16v-24a8 8 0 0 0 -8-8h-24v-16h24a8 8 0 0 0 8-8v-24h16v24a8 8 0 0 0 8 8h24z"/><path d="m104 328h-56a8 8 0 0 0 -8 8v64a8 8 0 0 0 8 8h56a8 8 0 0 0 8-8v-64a8 8 0 0 0 -8-8zm-8 64h-40v-48h40z"/><path d="m224 328h-88a8 8 0 0 0 0 16h88a8 8 0 0 0 0-16z"/><path d="m224 360h-88a8 8 0 0 0 0 16h88a8 8 0 0 0 0-16z"/><path d="m176 392h-40a8 8 0 0 0 0 16h40a8 8 0 0 0 0-16z"/></g></svg>
```
Comment (juliamitchelmore) [inline src/Components/AddMediaIcon/photo.svg]: Check the file name, we want to be descriptive :)

File: src/Components/AddMediaIcon/AddMediaIcon.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+          <p> + Add Files </p>
+        </div>
+      </label>
+      <input
+        hidden
+        multiple
+        id="fileUpload"
+        type="file"
+        accept="video/* image/*"
+        onChange={handleVideoUpload}
```
Comment (juliamitchelmore) [inline src/Components/AddMediaIcon/AddMediaIcon.tsx]: naming, this handles more than just videos

**Review by juliamitchelmore — APPROVED** [review]
Most of this looks fine, we'll loop back around on how best to handle individual video durations vs total video duration

**Comment by liviaw** [comment]
- change let to const
- handlevideoupload is now handlemediaupload
- for loop is now Array.from().map
- duration is included in mediapresenter (with no state yet)

**Pull request merged**